File: Sorting/869. Reordered Power of 2.py
class Solution:
    def reorderedPowerOf2(self, n: int) -> bool:
        target = [ch for ch in str(n)]
        target.sort()
        for i in range(33):
            s = [ch for ch in str(pow(2, i))]
            s.sort()
            if s == target:
                return True
        return False

File: 869. Reordered Power of 2.cpp
class Solution {
public:
    bool reorderedPowerOf2(int n)
    {
        string target = to_string(n), temp;
        sort(target.begin(), target.end());
        for (int i = 0; i <= 32; i++)
        {
            long long val = pow(2, i);
            temp = to_string(val);
            sort(temp.begin(), temp.end());
            if (temp == target)
                return true;
        }
        return false;
    }
};